### PR TITLE
MDEXP-256 Modify status of the completed export job

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**
https://issues.folio.org/browse/MDEXP-256

## Purpose
Currently when the job completes and the export file is generated the job status is set to Success. However, sometimes the file is created but only a part of the export succeeds due to some records erroring out.

For the jobs that exported successfully are the records set the status to: Completed
For the jobs that exported with some errors to: Completed with errors

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by illia-borysenko — APPROVED** [review]

**Comment by KVupp** [comment]
Please post the results if the data-migration script is a success, I think this will be a good candidate for using the racncher environment for testing

File: src/test/java/org/folio/rest/impl/EndToEndTest.java [hunk 1]
```diff
@@ -279,7 +279,7 @@ private void givenSetSourceStorageMockToReturnRecords() throws IOException {
   private Future<Object> assertSuccessJobExecution(TestContext context, FileDefinition fileDefinition,  Optional<JobExecution> jobExecutionOptional, Integer currentNumber, String total) {
     JobExecution jobExecution = jobExecutionOptional.get();
     context.assertTrue(isFileNameContainsJobExecutionHrId(new File(fileDefinition.getSourcePath()).getName(), jobExecution.getHrId()));
-    context.assertEquals(jobExecution.getStatus(), SUCCESS);
+    context.assertEquals(jobExecution.getStatus(), COMPLETED);
```
Comment (KVupp) [inline src/test/java/org/folio/rest/impl/EndToEndTest.java:282]: Please add a test case for "completed with errors"

**Comment by KVupp** [comment]
Just a reminder,  that this needs to be merged along with the UI story, to prevent the reference environments from breaking

**Comment by andrii-novytskyi** [comment]
Please, verify migration scripts and add confirmations to the PR

**Comment by Igor-Gorchakov** [comment]
> Please post the results if the data-migration script is a success, I think this will be a good candidate for using the rancher environment for testing

Hi @KVupp , rancher env doesn't run migration script while doing module-deploy, maybe that's because older version of module needs to be specified somewhere. I decided to verify the script on my localhost, doing module upgrate from v2.2.0-SNAPSHOT to v2.3.0.

**0. deploy mod-data-export v2.2.0**
image

**1 create jobs v2.2.0**
image

**2 install mod-data-export v2.3.0**
image

**3 check install logs v2.3.0**
image

**4 check jobs in database v2.3.0**
image

**Comment by KVupp** [comment]
@Igor-Gorchakov  Thanks for verifying in your local. And bringing it to my attention. Let's discuss if we can figure out a way to use rancher environment for that purpose

File: src/test/java/org/folio/rest/impl/EndToEndTest.java (hunk 1 above)
Comment (KVupp) [reply src/test/java/org/folio/rest/impl/EndToEndTest.java:282]: Please take care of this @Igor-Gorchakov

Comment (Igor-Gorchakov) [reply src/test/java/org/folio/rest/impl/EndToEndTest.java:282]: The test is in place

Comment (KVupp) [reply src/test/java/org/folio/rest/impl/EndToEndTest.java:282]: Thanks!

**Pull request merged**